src/traceViewer/web/index.tsx

```diff
@@ -20,7 +20,28 @@ import './third_party/vscode/codicon.css';
 import { Workbench } from './ui/workbench';
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
-import * as monaco from 'monaco-editor';
+import { ActionTraceEvent } from '../traceTypes';
+
+declare global {
+  const monaco: typeof import('monaco-editor');
+  interface Window {
+    getTraceModel(): Promise<TraceModel>;
+    readFile(filePath: string): Promise<string>;
+    renderSnapshot(action: ActionTraceEvent): void;
+  }
+};
+
+async function loadMonaco() {
+  // Note: it is important to use global eval and require to not conflict with webpack.
```

Comment:
Further explaining for future joel: webpack doesn't touch eval. However it changes our code so that this === undefined like we are in a module. That is propagated into the eval. So we use an indirect eval to force the global this to be used.